Tests/FunctionalKitTests/OptionalTests.swift

```diff
@@ -0,0 +1,20 @@
+import XCTest
+import Operadics
+@testable import FunctionalKit
+
+class OptionalTests: XCTestCase {
+    
+    func testLift() {
```

Comment:
If this kind of test is OK, I will create tests for all types. Otherwise we can use Property based testing
CC: @broomburgo @ChrisMyrants @josete89